lib/livebook/web_socket/client.ex

```diff
@@ -0,0 +1,197 @@
+defmodule Livebook.WebSocket.Client do
+  @moduledoc false
+
+  alias Mint.WebSocket.UpgradeFailureError
+
+  @ws_path "/livebook/websocket"
+
+  @type conn :: Mint.HTTP.t()
+  @type websocket :: Mint.WebSocket.t()
+  @type frame :: :close | {:binary, binary()}
+  @type ref :: Mint.Types.request_ref()
+  @type ws_error :: Mint.WebSocket.error()
+  @type mint_error :: Mint.Types.error()
+
+  @type connect_result ::
+          {:ok, conn(), ref()}
+          | {:error, mint_error()}
+          | {:error, conn(), ws_error()}
+
+  @type receive_result ::
+          {:ok, conn(), Response.t() | :connect}
+          | {:error, conn(), Response.t()}
+          | {:error, conn(), :unknown}
+
+  @type send_result ::
+          {:ok, conn(), websocket()}
+          | {:error, conn() | websocket(), term()}
```

Comment:
We only reference those in a single place, so probably fine to inline and keep closer to where it's relevant?

Reply:
Yeah, there's no point anymore to use this one